### DISCUSSION TITLE
Fetch watcher delay

### DISCUSSION THREAD

**Discussion by hecktarzuli** [nuxt/nuxt#7956] [General]
I'm  wondering why it takes 100-150ms to trigger a watcher when fetch is complete. (in SSR or classic static mode)

Here, I have a console.log right after the api response in my fetch().  In the same component I'm watching `$fetchState.pending` and `$nuxt.nbFetching`.  Both take a long time to update.  I was hoping to leverage these watchers to trigger performance metrics (Speedcurve), but with such a high delay it wouldn't be accurate.

![image](https://user-images.githubusercontent.com/2112282/90971695-226f2680-e4d8-11ea-9061-22945a63e1bd.png)

@danielroe @pi0 @Atinux

**Comment by atinux** [comment]
Could you try if this.$options.fetch is defined?

*Sébastien CHOPIN*
CTO & Co-founder of NuxtJS

( https://nuxtjs.org/ )

*Phone:* +33 (0)7 80 90 41 62
*Email:* seb@nuxtjs.com
*Location* : 5 Allée de Tourny, 33000 Bordeaux. FRANCE
RCS Bordeaux - 828 193 722 - FR92828193722

**Reply by hecktarzuli** [reply]
Sorry, I think you may have answered a cached/old question.  I ditched the 'fetch' detection (and yeah I did try this.$options.fetch).. 

Now I'm focusing on why there is such a large delay in fetch completion and the update of the fetch watcher.  I did try this same test with a prod build and had the same results.

At this point I'm just going to close this discussion, thanks!
